openedx/core/djangoapps/xblock/rest_api/views.py

```diff
@@ -168,3 +176,94 @@ def cors_allow_xblock_handler(sender, request, **kwargs):  # lint-amnesty, pylin
 
 
 check_request_enabled.connect(cors_allow_xblock_handler)
+
+
+@view_auth_classes()
+class BlockFieldsView(APIView):
+    """
+    View to get/edit a v2 xblock as json
+
+    This class mimics the functionality of xblock_handler in block.py (for v1 xblocks), but for v2 xblocks.
+    However, it only implements the exact subset of functionality needed to support the v2 editors (from
+    the frontend-lib-content-components project). As such, it only supports GET and POST, and only the
+    POSTing of data/metadata fields.
+    """
+
+    @atomic
+    def get(self, request, usage_key_str):
+        """
+        retrieves the xblock, returning display_name, data, and metadata
+        """
+        try:
+            usage_key = UsageKey.from_string(usage_key_str)
+        except InvalidKeyError as e:
+            raise NotFound(invalid_not_found_fmt.format(usage_key=usage_key_str)) from e
+
+        block = load_block(usage_key, request.user)
+        block_dict = {
+            "display_name": get_block_display_name(block),
+            "data": block.data,
+            "metadata": block.get_explicitly_set_fields_by_scope(Scope.settings),
```

Comment:
What about the `Scope.content` fields - are they not needed by any of these editors? Or just `data` is the only Scope.content field needed?

Reply:
we only expect those three fields, and `data` is the only `Scope.content` one that we use.